### PR TITLE
feat(base-map): Add support for i18n on map controls

### PR DESCRIPTION
**Pull request by amy-corson-ibigroup**
Passes translated strings to MapLibre to override their default map controls.

### CODE REVIEW

File: packages/base-map/src/mapControlLocale.tsx
Comment (binh-dam-ibigroup) [inline packages/base-map/src/mapControlLocale.tsx:31]:
```suggestion
): Record<string, string> => {
```

If you decide to use the suggested syntax, you might need to cast types in the function body.

File: packages/base-map/i18n/en-US.yml
Comment (binh-dam-ibigroup) [inline packages/base-map/i18n/en-US.yml:3]: Do we use all of these settings, and therefore do we need to support i18n for all of these?

File: packages/base-map/src/mapControlLocale.tsx
Comment (binh-dam-ibigroup) [inline packages/base-map/src/mapControlLocale.tsx:36]: Clever. Add a string group in the exceptions file in the i18n folder for this package (see #589).

**Review by binh-dam-ibigroup — COMMENTED** [review]
Do we need all of these strings?

**Comment by amy-corson-ibigroup** [comment]
Good point about the strings, I've removed the ones I'm pretty sure we don't use!

File: packages/base-map/i18n/en-US.yml
Comment (binh-dam-ibigroup) [inline packages/base-map/i18n/en-US.yml:11]: Where does the scale control get used? I don't see a scale on most maps.

File: packages/base-map/i18n/i18n-exceptions.json
Comment (binh-dam-ibigroup) [inline packages/base-map/i18n/i18n-exceptions.json:8]: For this one, just
```suggestion
    "otpUi.mapControls.NavigationControl.*": [
```

File: packages/base-map/i18n/en-US.yml
Comment (amy-corson-ibigroup) [reply packages/base-map/i18n/en-US.yml:11]: This is in the "with optional controls" story, but we can always remove that scale from the map if it's not representative of how we use it?

**Review by binh-dam-ibigroup — APPROVED** [review]

File: packages/base-map/i18n/i18n-exceptions.json
Comment (binh-dam-ibigroup) [inline packages/base-map/i18n/i18n-exceptions.json:13]:
```suggestion
    "otpUi.mapControls.ScaleControl.*": [
```

File: packages/base-map/i18n/i18n-exceptions.json
Comment (binh-dam-ibigroup) [inline packages/base-map/i18n/i18n-exceptions.json:3]: Here, do
either `otpUi.mapControls.CooperativeGesturesHandler.*`
or
```
  "Mac",
  "Mobile",
  "Windows"
```

**Review by binh-dam-ibigroup — CHANGES_REQUESTED** [review]
Some suggestions to make i18n checks pass.

**Comment by amy-corson-ibigroup** [comment]
Thanks for the help and the French translations, @binh-dam-ibigroup!

**Review by binh-dam-ibigroup — APPROVED** [review]

File: packages/base-map/i18n/i18n-exceptions.json
Comment (miles-grant-ibigroup) [inline packages/base-map/i18n/i18n-exceptions.json:1]: It's not clear why this file has to be here. I think I know but it might be helpful to add to the readme or a comment somewhere

**Review by miles-grant-ibigroup — CHANGES_REQUESTED** [review]
A few nitpicks

File: packages/base-map/i18n/i18n-exceptions.json
Comment (miles-grant-ibigroup) [reply packages/base-map/i18n/i18n-exceptions.json:1]: Gorgeous. Thanks so much

**Comment by binh-dam-ibigroup** [comment]
@amy-corson-ibigroup Let me add partial translations as requested before merging.

**Review by miles-grant-ibigroup — APPROVED** [review]
Looks great!

**Comment by amy-corson-ibigroup** [comment]
@binh-dam-ibigroup @miles-grant-ibigroup translations have been provided. I'll merge today provided everything looks okay!

**Pull request merged**
